File: backend/app/schemas/formula.py
from typing import List, Optional

from pydantic import BaseModel


class FormulaResponse(BaseModel):
    id: str
    subject: str
    category: Optional[str]
    name: str
    latex: str
    description: Optional[str]
    grade_levels: List[str]

    class Config:
        from_attributes = True


class FormulaDetailResponse(FormulaResponse):
    keywords: List[str]
    related_formulas: List[FormulaResponse]


class SaveFormulaRequest(BaseModel):
    name: str
    latex: str
    subject: Optional[str] = "math"
    description: Optional[str] = None

File: mobile/src/types/formula.ts
import { Subject } from './scan';

export interface Formula {
  id: string;
  subject: Subject;
  category?: string;
  name: string;
  latex: string;
  description?: string;
  gradeLevels: string[];
}

export interface FormulaDetail extends Formula {
  keywords: string[];
  relatedFormulas: Formula[];
}

export interface FormulaSearchParams {
  subject?: Subject;
  category?: string;
  gradeLevel?: string;
  keyword?: string;
  page?: number;
  limit?: number;
}
